xmtp/src/client.rs

```diff
@@ -32,6 +35,8 @@ pub enum ClientError {
     PublishError(String),
     #[error("storage error: {0}")]
     Storage(#[from] StorageError),
+    #[error("dieselError: {0}")]
+    Ddd(#[from] diesel::result::Error),
```

Comment:
With the client now performing transactions directly, having diesel errors be converted directly to ClientErrors is easier. Open to another Error handing strategy

Reply:
I don't see why it wouldn't be possible, however that would not be my current preference. I hear your desire to have a central data management interface. If going that route I'd prefer that a separate object be built for that purpose, rather than creating monolithic structs.

A Two Struct approach would be:
StorageObject: (EncryptedStore -> SqlcipherStore)
QueryManager:

Separating the two would allow ownership of the underlying storageObject to be more flexible, and better facilitate testing. 

Very open to discussion on this